pkg/pgmodel/querier/query_builder.go

```diff
@@ -242,7 +242,29 @@ func canAttemptPushdown(metadata *promqlMetadata) bool {
 	path := metadata.path // PromQL AST.
 	queryHints := metadata.queryHints
 	selectHints := metadata.selectHints
-	return extension.ExtensionIsInstalled && queryHints != nil && !hasSubquery(path) && selectHints != nil
+
+	// We can't push down without hints
+	if queryHints == nil || selectHints == nil {
+		return false
+	}
+
+	// We can't do pushdowns without the extension
+	if !extension.ExtensionIsInstalled {
+		return false
+	}
+
+	// We can't handle subqueries in pushdowns
+	if hasSubquery(path) {
+		return false
+	}
+
+	// We can't handle offsets in pushdowns
+	vs, isVectorSelector := queryHints.CurrentNode.(*parser.VectorSelector)
+	if isVectorSelector && (vs.Offset != 0 || vs.OriginalOffset != 0) {
```

Comment:
Base on Prometheus docs `OriginalOffset` is actual offset set from the query so it seems no need to check `Offset` as that is just derived during query execution.

Reply:
done

Reply:
The plot thickens: this change was not good. We do actually need to validate that both `Offset` and `OriginalOffset` are zero.